static/src/javascripts/projects/common/modules/commercial/creatives/revealer.js

```diff
@@ -0,0 +1,30 @@
+define([
+    'common/utils/fastdom-promise',
+    'common/utils/template',
+    'common/modules/commercial/creatives/add-tracking-pixel',
+    'text!common/views/commercial/creatives/revealer.html'
+], function(fastdom, template, addTrackingPixel, revealerStr) {
+    var revealerTpl;
+
+    function Revealer($adSlot, params) {
+        revealerTpl || (revealerTpl = template(revealerStr));
+
+        return Object.freeze({
```

Comment:
I meant more; why depend on hoisting? If so you could just put this at the very top of the `function Revealer`?